### PR TITLE
Improve Linux build

### PR DESCRIPTION
**Pull request by natemcmaster**
#### For Travis CI:
 - Create build script for Travis CI. Due to bugs in Mono 4.2.0, run only on CoreCLR.
 - Fix file location of logging file
 - Add explicit reference to System.Net.Security

#### Enabling better Mono build/testing
 - Downgrade all test projects to .NET 4.5.1
 - Skip tests that require external SQL Server (if not configured)
 - Remove PDB symbol compliation from RoslynCompilationService. (Only affects scaffolding E2E tests)
 - Add dispose calls to all use of SqliteTestStore. See #3749. Doesn't fix as there are bugs in xunit/mono.
 - Add platform helpers to testing

Initial progress towards #3594

### CODE REVIEW

**Comment by natemcmaster** [comment]
cc @rowanmiller I changed the implementation of "AdventureWorksDatabaseBenchmark"
cc @Eilon can we enable Travis builds?

File: test/EntityFramework.Core.FunctionalTests/TestFileLogger.cs [hunk 1]
```diff
@@ -36,8 +36,8 @@ public void Dispose()
 
         protected TestFileLogger(string fileName = "data-test.log")
         {
-            var logDirectory
-                = Path.Combine(Environment.ExpandEnvironmentVariables("%USERPROFILE%"), ".klog");
+            var homeDir = Environment.GetEnvironmentVariable("USERPROFILE") ?? Environment.GetEnvironmentVariable("HOME");
+            var logDirectory = Path.Combine(homeDir, ".klog");
```
Comment (smitpatel) [inline test/EntityFramework.Core.FunctionalTests/TestFileLogger.cs]: Do we have any reason for using `.klog` folder? It existed during the `k` world but after renaming to `dnx` world, `.klog` is not used by anything else.

Comment (natemcmaster) [inline test/EntityFramework.Core.FunctionalTests/TestFileLogger.cs]: Idk. It appears other @aspnet projects don't use `.klog` anymore. What is this log file used for anyways?

Comment (anpete) [inline test/EntityFramework.Core.FunctionalTests/TestFileLogger.cs]: IIRC, this was an old, primitive way of observing test output. Can likely be deleted now.

**Comment by Eilon** [comment]
Sorry for the delay - I just enabled Travis CI for this repo.

File: test/EntityFramework.Core.Tests/DbContextTest.cs [hunk 2]
```diff
@@ -2579,7 +2580,10 @@ public void It_throws_with_derived_name()
             context.Dispose();
 
             var ex = Assert.Throws<ObjectDisposedException>(() => context.Model);
-            Assert.Contains(nameof(EarlyLearningCenter), ex.Message);
+            if (!TestPlatformHelper.IsMono)
```
Comment (lajones) [inline test/EntityFramework.Core.Tests/DbContextTest.cs]: Is this temporary? If so add TODO. If not why is the exception message different on Mono?

**Comment by lajones** [comment]
Looks good baring minor comment. :shipit:

File: test/EntityFramework.Core.Tests/DbContextTest.cs (hunk 2 above)
Comment (natemcmaster) [inline test/EntityFramework.Core.Tests/DbContextTest.cs]: Not temporary. Mono just did not implement this error message the same way.

Comment (lajones) [inline test/EntityFramework.Core.Tests/DbContextTest.cs]: Hmmmm. OK. Guess there's nothing we can do about that. Go ahead and check in.

**Pull request merged**